src/khoj/routers/api.py

```diff
@@ -553,6 +556,54 @@ async def chat_options(
     return Response(content=json.dumps(cmd_options), media_type="application/json", status_code=200)
 
 
+@api.post("/speak")
+@requires(["authenticated"])
+async def transcribe_audio(request: Request, common: CommonQueryParams, file: UploadFile = File(...)):
+    user: KhojUser = request.user.object
+    audio_filename = f"{user.uuid}-{str(uuid.uuid4())}.webm"
+    user_message: str = None
+
+    # Transcribe the audio from the request
+    try:
+        # Store the audio from the request in a temporary file
+        audio_data = await file.read()
+        with open(audio_filename, "wb") as audio_file_writer:
+            audio_file_writer.write(audio_data)
+        audio_file = open(audio_filename, "rb")
+
+        # Send the audio data to the Whisper API
+        speech_to_text_config = await ConversationAdapters.get_speech_to_text_config()
+        openai_chat_config = await ConversationAdapters.get_openai_chat_config()
+        if not openai_chat_config or not speech_to_text_config:
+            # If the user has not configured a speech to text model, return an unprocessable entity error
+            status_code = 422
+        elif speech_to_text_config.model_type == ChatModelOptions.ModelType.OPENAI:
```

Comment:
What happens if the `model_type` is offline? Should we load the whisper model directly in code?

I think this one also makes sense to abstract into separate files, perhaps under `search/khoj/speech` or equivalent? Then add one file for handling offline and one for online, similar to what we do with `chat`.